core/parentchain/light-client/src/lib.rs

```diff
@@ -30,28 +30,23 @@ extern crate sgx_tstd as std;
 pub use finality_grandpa::BlockNumberOps;
 pub use sp_finality_grandpa::{AuthorityList, SetId};
 
-use crate::state::ScheduledChangeAtBlock;
-use codec::{Decode, Encode};
-use core::iter::Iterator;
 use error::Error;
-use finality_grandpa::voter_set::VoterSet;
 use itp_ocall_api::EnclaveOnChainOCallApi;
-use itp_storage::{Error as StorageError, StorageProof, StorageProofChecker};
-use justification::GrandpaJustification;
-use log::*;
-use sp_finality_grandpa::{
-	AuthorityId, AuthorityWeight, ConsensusLog, ScheduledChange, GRANDPA_ENGINE_ID,
-};
+use itp_storage::StorageProof;
+use sp_finality_grandpa::{AuthorityId, AuthorityWeight, ConsensusLog, GRANDPA_ENGINE_ID};
 use sp_runtime::{
-	generic::{Digest, OpaqueDigestItemId},
-	traits::{Block as ParentchainBlockTrait, Hash as HashTrait, Header as HeaderTrait},
-	Justification, Justifications, OpaqueExtrinsic,
+	generic::{Digest, OpaqueDigestItemId, SignedBlock},
+	traits::{Block as ParentchainBlockTrait, Header as HeaderTrait},
+	Justifications, OpaqueExtrinsic,
 };
-use state::RelayState;
-use std::{collections::BTreeMap, fmt, vec::Vec};
+use std::vec::Vec;
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::grandpa_light_validation::GrandpaLightValidation;
```

Comment:
What is the reason for this `sgx` feature restriction?

Reply:
now it seems to work without the restriction